interoperator/pkg/controller/multiclusterdeploy/sfservicebindingreplicator/sfservicebindingreplicator_controller.go

```diff
@@ -0,0 +1,360 @@
+		Client:          mgr.GetClient(),
+		scheme:          mgr.GetScheme(),
+		clusterRegistry: clusterRegistry,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("sfservicebindingreplicator-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to SFServiceInstanceReplicator
+	err = c.Watch(&source.Kind{Type: &osbv1alpha1.SFServiceBinding{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	watchEvents, err := getWatchChannel("sfservicebindings")
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Channel{Source: watchEvents}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileSFServiceBindingReplicator{}
+
+// ReconcileSFServiceBindingReplicator reconciles a ReconcileSFServiceBindingReplicator object
+type ReconcileSFServiceBindingReplicator struct {
+	client.Client
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile reads that state of the cluster for a SFServiceInstanceReplicator object and makes changes based on the state read
+// and what is in the SFServiceInstanceReplicator.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  The scaffolding writes
+// a Deployment as an example
+// +kubebuilder:rbac:groups=osb.servicefabrik.io,resources=*,verbs=*
+func (r *ReconcileSFServiceBindingReplicator) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the SFServiceInstanceReplicator instance
+	binding := &osbv1alpha1.SFServiceBinding{}
+	replica := &osbv1alpha1.SFServiceBinding{}
+	err := r.Get(context.TODO(), request.NamespacedName, binding)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	bindingID := binding.GetName()
+	state := binding.GetState()
+
+	clusterID, err := binding.GetClusterID(r)
+	if err != nil {
+		log.Info("clusterID not set. Ignoring", "instance", bindingID)
+		return reconcile.Result{}, nil
+	}
+
+	if clusterID == constants.MasterClusterID {
+		// Target cluster is mastercluster itself
+		// Replication not needed
+		log.Info("Target cluster is master cluster itself, replication not needed..")
+		return reconcile.Result{}, nil
+	}
+	log.Info("Binding replication started for sister cluster:", "bindinID", bindingID, "clusterID", clusterID)
+
+	targetClient, err := r.clusterRegistry.GetClient(clusterID)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if state == "delete" && !binding.GetDeletionTimestamp().IsZero() {
```

Comment:
Should we reconcile the namespace here as well to avoid running into any race conditions?

Reply:
we assume namespace is already there when a new binding is created. This is valid because a binding can be created only after an instance is created. If we reconile for ns for bindings also it creates problems in deltion of ns

Reply:
Yes. But it could be that event by the `ServiceInstance` controller is not yet processed. This is what I meant by the race condition.